File: test-web-app/src/main/java/com/example/app/service/TestUserService.java
package com.example.app.service;

import com.example.app.data.TestUser;
import com.example.app.repository.TestUserRepository;
import org.springframework.stereotype.Service;
import ru.nsu.manasyan.kcache.core.annotations.KCacheEvict;

import java.util.List;

@Service
public class TestUserService {
    private final TestUserRepository repository;

    public TestUserService(TestUserRepository userRepository) {
        this.repository = userRepository;
    }

    public List<TestUser> getUsers() {
        return repository.findAll();
    }

    @KCacheEvict(tables = {"users"})
    public void addUser(TestUser user) {
        repository.save(user);
    }

    public TestUser getUser(String id) {
        return repository.findUserById(id);
    }

    @KCacheEvict(tables = {"users"})
    public void evictUsersCache() {}
}

File: test-web-app/src/main/kotlin/com/example/app/service/TestUserService.kt
package com.example.app.service

import com.example.app.data.TestUser
import com.example.app.repository.TestUserRepository
import org.springframework.stereotype.Service
import ru.nsu.manasyan.kcache.core.annotations.KCacheEvict
import java.util.*

@Service
class TestUserServiceKt(
    private val repository: TestUserRepository
) {
    fun getUsers(): List<TestUser> {
        return repository.findAll()
    }

    @KCacheEvict(tables = ["users"])
    fun addUser(user: TestUser) {
        repository.save(user)
    }

    fun generateUser() {
        repository.save(
            TestUser(
                UUID.randomUUID().toString(),
                UUID.randomUUID().toString(),
                777
            )
        )
    }

    @KCacheEvict(tables = ["users"])
    fun evictUsersCache() {
    }
}